ui/components/providers/workflow/forms/select-credentials-type/gcp/via-service-account-form.tsx

```diff
@@ -0,0 +1,160 @@
+import { GCPServiceAccountKeyForm } from "./credentials-type/gcp-service-account-key-form";
+
+export const ViaServiceAccountForm = ({
+  searchParams,
+}: {
+  searchParams: { type: string; id: string };
+}) => {
+  const router = useRouter();
+  const { toast } = useToast();
+  const searchParamsObj = useSearchParams();
+
+  // Handler for back button
+  const handleBackStep = () => {
+    const currentParams = new URLSearchParams(window.location.search);
+    currentParams.delete("via");
+    router.push(`?${currentParams.toString()}`);
+  };
+
+  const providerType = searchParams.type;
+  const providerId = searchParams.id;
+
+  const formSchema = addCredentialsServiceAccountFormSchema(providerType);
+  type FormSchemaType = z.infer<typeof formSchema>;
+
+  const form = useForm<FormSchemaType>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      providerId,
+      providerType,
+      ...(providerType === "gcp"
+        ? {
+            service_account_key: "",
+            secretName: "",
+          }
+        : {}),
+    },
+  });
+
+  const isLoading = form.formState.isSubmitting;
+
+  const onSubmitClient = async (values: FormSchemaType) => {
+    const formData = new FormData();
+
+    Object.entries(values).forEach(([key, value]) => {
+      if (value !== undefined && value !== "") {
+        formData.append(key, String(value));
+      }
+    });
+
+    try {
+      const data = await addCredentialsProvider(formData);
+
+      if (data?.errors && data.errors.length > 0) {
+        data.errors.forEach((error: ApiError) => {
+          const errorMessage = error.detail;
+
+          switch (error.source.pointer) {
+            case "/data/attributes/secret/service_account_key":
+              form.setError("service_account_key" as keyof FormSchemaType, {
+                type: "server",
+                message: errorMessage,
+              });
+              break;
+            default:
+              toast({
+                variant: "destructive",
+                title: "Oops! Something went wrong",
+                description: errorMessage,
+              });
+          }
+        });
+      } else {
+        router.push(
+          `/providers/test-connection?type=${providerType}&id=${providerId}`,
+        );
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error("Error during submission:", error);
+      toast({
+        variant: "destructive",
+        title: "Submission failed",
+        description: "An error occurred while processing your request.",
+      });
+    }
+  };
+
+  return (
+    <Form {...form}>
+      <form
+        onSubmit={form.handleSubmit(onSubmitClient)}
+        className="flex flex-col space-y-4"
+      >
+        <input type="hidden" name="providerId" value={providerId} />
+        <input type="hidden" name="providerType" value={providerType} />
+
+        <ProviderTitleDocs providerType={providerType as ProviderType} />
```

Comment:
This casting is not needed as the providerType prop inside ProviderTitleDocs is nicely typed.
```suggestion
        <ProviderTitleDocs providerType={providerType} />
```